apps/ui/src/components/EditorVote.vue

```diff
@@ -0,0 +1,128 @@
+<script setup lang="ts">
+import Draggable from 'vuedraggable';
+import { BASIC_CHOICES } from '@/helpers/constants';
+import { Draft, VoteType } from '@/types';
+
+const proposal = defineModel<Draft>({ required: true });
+
+defineProps<{
+  error?: string;
+  votingTypes: VoteType[];
+}>();
+
+const VOTING_TYPES_INFO = {
+  basic: {
+    label: 'Basic voting',
+    description: 'Single choice voting with three choices: For, Against or Abstain.'
+  },
+  'single-choice': {
+    label: 'Single choice voting',
+    description: 'Each voter may select only one choice.'
+  },
+  approval: {
+    label: 'Approval voting',
+    description: 'Each voter may select any number of choices.'
+  }
+};
+
+function handleVoteTypeSelected(type: VoteType) {
+  if (!proposal.value) return;
+
+  proposal.value.type = type;
+
+  if (proposal.value.type === 'basic') {
+    proposal.value.choices = [...BASIC_CHOICES];
+  }
+}
+</script>
+
+<template>
+  <template v-if="votingTypes.length > 1 || votingTypes[0] !== 'basic'">
+    <div class="s-base mb-5">
+      <h4 class="eyebrow mb-2.5">Voting type</h4>
+      <div class="flex flex-col gap-[12px]">
+        <button
+          v-for="(type, index) in votingTypes"
+          :key="index"
+          class="border rounded-lg p-2.5 flex gap-3 text-left"
+          :class="{ 'border-skin-text bg-skin-input-bg': proposal.type === type }"
+          @click="handleVoteTypeSelected(type as VoteType)"
+        >
+          <div
+            class="h-[82px] w-[122px] hidden sm:block rounded-lg shrink-0 bg-skin-active-bg"
+          ></div>
+          <div class="grow">
+            <span class="text-skin-heading">{{ VOTING_TYPES_INFO[type].label }}</span>
+            <div>
+              {{ VOTING_TYPES_INFO[type].description }}
+            </div>
+          </div>
+          <div class="w-[20px] text-right">
+            <IH-check v-if="proposal.type === type" />
+          </div>
+        </button>
+      </div>
+    </div>
+
+    <div class="s-base">
+      <h4 class="eyebrow mb-2.5">Choices</h4>
+      <div class="flex flex-col gap-[10px]">
+        <Draggable
+          v-model="proposal.choices"
+          handle=".handle"
+          class="flex flex-col gap-[10px]"
+          item-key="index"
+        >
+          <template #item="{ index }">
+            <div>
+              <div
+                class="flex border items-center rounded-lg bg-skin-input-bg h-[40px] gap-[12px] pl-2.5"
+                :class="{ 'border-skin-danger': error && index === 0 }"
+              >
+                <div
+                  class="text-skin-link opacity-50"
+                  :class="{
+                    'handle hover:opacity-100 cursor-grab': proposal.type !== 'basic',
+                    'cursor-not-allowed': proposal.type === 'basic'
+                  }"
+                >
+                  <IH-switch-vertical />
+                </div>
+                <div class="grow">
+                  <input
```

Comment:
I think when you change voting type from basic to another it should reset the state instead of keeping the same basic voting choices. Also we should show placeholder as in Figma, the `(optional)` was just for snapshot.org because we didn't have a way to delete a choice

Reply:
And what other to single to approval ? Should we also reset the choices ? Or should we keep a "draft" for each choices type, so when the user select back anothe voting type, it fills back the previous choices ?

For the placeholder, maybe "Type in a choice" ? Should be clear and concise enough to tell that the empty choice will be ignored, so people do not wonder where does his empty choice go

Reply:
I think we should only reset when switching from Basic to any other type.

To me placeholder should be `Choice 1` and so forth

Reply:
![Screenshot 2024-02-25 at 16 04 24](https://github.com/snapshot-labs/sx-monorepo/assets/495709/3725bf27-fab9-4005-88d9-bed271d0b257)

What about the placeholder on error ?

Reply:
The error is not ideal as it should say "At least one choice required" and it should only show when publish is clicked. Publish button should always be enabled and error should only show when it's was clicked 

I also noticed when I change the choices and then switch to Basic it still shows the choices I entered which is a bug

Reply:
The choice input should also validate the length of the choice string, now it seems like I can enter longer string than 32

Reply:
Fixed, it will not show any form error on choices error, will only disable the "Publish" button

Reply:
This is a limitation of ajv validation, which only validate the first choice.